azure-devops/azext_devops/dev/boards/relations.py

```diff
@@ -0,0 +1,107 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from knack.util import CLIError
+from azext_devops.devops_sdk.v5_0.work_item_tracking.models import JsonPatchOperation, Wiql
+
+from azext_devops.dev.common.services import (get_work_item_tracking_client,
+                                              resolve_instance)
+
+logger = get_logger(__name__)
+
+
+def get_relation_types_show(organization=None, detect=None):
+    """ List work item relations supported in the organization.
+    """
+    organization = resolve_instance(detect=detect, organization=organization)
+    client = get_work_item_tracking_client(organization)
+    return client.get_relation_types()
+
+
+def add_relation(id, relation_type, target_ids, organization=None, detect=None):  # pylint: disable=redefined-builtin
```

Comment:
Table output is not very helpful. We should list all related items with relation type.
azure-devops> az boards work-item relation add --id 103 --relation-type 'Duplicate of' --target-ids 104
![image](https://user-images.githubusercontent.com/11888714/56341278-28614300-61d2-11e9-81c4-98bf9e712825.png)

Reply:
Relations are array so that can grow, I really don't think anyone should be trying to consume the output as table if thy want to get relations

Reply:
Can't we have the same terminology -? ID, Type Title Target ID Relationship Type? What is stopping us from providing a useful table output? Is it only the assumption that no one would use table?

Reply:
Plus - where are you getting this wheel from? 🤔

Reply:
well there are multiple reasons why this is should not be done.
1. say a WI has 20 relations (same type)
then the `Target ID` column will grow and given that they are distinct IDs and not same name so trimming that info is not correct
2. say a WI has 5 relations of 4 different type
How will be present this info in one column? 
Only possible way will be to show multiple columns for same work item, but that will give false impression to user that there are multiple work items

wheel is present in the PR build as artifact

If at all this seems like something which user wants to know I can suggest we show relations_count in the table

Reply:
ahh, got it now
we can do that

Reply:
now I got the requirement.
maybe this is a valid case.

but then should we also have a show for relations? because someone just want to see relation without add/remove?

Reply:
e.g. In build tag add or build tag list we only list down all tags and no other information. 
Here we can list artifacts and their relationship type. 

Where I am getting the wheel from.. 
```hub pr checkout 541```
And invoke from inside the path **azure-devops>** I do not even need to build the wheel :)

Reply:
done, but still stuck with showing Url because that is what we get in response from service and given that it can be any Url, I am trying not to manipulate it

![image](https://user-images.githubusercontent.com/43203065/56349120-85b2bf80-61e5-11e9-96f9-f76bd3ea32b0.png)

Reply:
Fair enough for first try